src/app/parcels/form/ParcelForm.tsx

```diff
@@ -230,36 +230,16 @@ const ParcelForm: React.FC<ParcelFormProps> = ({
             last_updated: fields.lastUpdated,
         };
 
-        if (editMode) {
-            const { error: updateParcelError } = await updateParcel(formToAdd, parcelId!);
-            if (updateParcelError) {
-                let errorMessage: string;
-                switch (updateParcelError.type) {
-                    case "failedToUpdateParcel":
-                        errorMessage = `Failed to update parcel. Log ID: ${updateParcelError.logId}`;
-                        break;
-                    case "concurrentUpdateConflict":
-                        errorMessage = `Record has been edited recently - please refresh the page. LogID: ${updateParcelError.logId}`;
-                        break;
-                }
-                setSubmitErrorMessage(errorMessage);
-                return;
-            }
-        } else {
-            const { error: insertParcelError } = await insertParcel(formToAdd);
-            if (insertParcelError) {
-                let errorMessage: string;
-                switch (insertParcelError.type) {
-                    case "failedToInsertParcel":
-                        errorMessage = `Failed to insert parcel. Log ID: ${insertParcelError.logId}`;
-                        break;
-                }
-                setSubmitErrorMessage(errorMessage);
-                return;
-            }
-        }
-        router.push("/parcels/");
+        const { error, parcelId } = await writeParcelInfoToDatabase(parcelRecord);
         setSubmitDisabled(false);
```

Comment:
Previously there was a return in line 258, before the setSubmitDisabled. This meant that when there was an error, the submit button gets permanently greyed out, and you can't click it. I think this might be a bug, and should be fixed now, but lmk if this is intended :)

Reply:
I think this was intended, but I don't really have a preference, @rickaw @minalui thoughts?

Reply:
It feels a little odd that you can't resubmit in the case of an error (without refreshing the page), especially a currency error, that waiting a bit and resubmitting should fix. That being said, if there's a reason am happy to remove line 228 (which will keep the submit button greyed out in the case of an error).

Reply:
Thanks for flagging this, originally this was intended so that users don't try to edit the record again until they have up to date data in the form (concurrent editing error). I suppose if we can distinguish between concurrent editing error (we don't allow resubmitting) and other errors (allows resubmitting)?

Reply:
Yeah I would be happy to implement this - I guess the error message does say please refresh the page so that makes sense

Reply:
Done :)